internal/server/server.go

```diff
@@ -189,12 +197,72 @@ func StartUp(Version string) {
 
 	// Start rest service
 	srvRest := createRestServer(conf.Config.Basic.RestIp, conf.Config.Basic.RestPort, conf.Config.Basic.Authentication)
+
+	var ln net.Listener
+	var lerr error
+	edgexCredentialFile := os.Getenv("EDGEX_CREDENTIALS")
```

Comment:
When will this env set?

Reply:
i have reworked how this works. please review. I have added four (4) new environment variables all in `internal/edgex/vault.go`:
* EDGEX_SERVICE_NAME
* EDGEX_CREDENTIALS
* EDGEX_VAULT_HOST
* EDGEX_VAULT_PORT

they are expected to be set when ekuiper starts -- likely from docker. i could adapt to a config file if you want/need/prefer. marking this as resolved...